keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -254,15 +264,17 @@ a `ServiceExport` in a cluster will signify that the `Service` with the same
 name and namespace as the export should be visible to other clusters in the
 supercluster.
 
-Another CRD called `ServiceImport` will be introduced to store information
-about the services exported from each cluster, e.g. topology. This is analogous
-to the traditional `Service` type in Kubernetes. Each cluster will have a
-coresponding `ServiceImport` for each uniquely named `Service` that has been
-exported within the supercluster, referenced by namespaced name.
+Another CRD called `ServiceImport` will be introduced to act as the in-cluster
+representation of a multi-cluster service in each importing cluster. This is
+analogous to the traditional `Service` type in Kubernetes. Each cluster will
```

Comment:
Is it a requirement that every cluster have an import for every super-service?  Or can the impl allow policies that control that?

Reply:
I think we probably need to allow scenarios where a given service may not be imported by every cluster.  These options have come up in discussion:

1. Namespace level _(recommended)_: if a namespace does not exist in a cluster, `ServiceImport`s in that namespace will not be created. This fits nicely with namespace sameness and a few other conversations I've had on this topic. It's easy to reason about and still allows a service to be isolated from some subset of clusters. I can reliably set expectations around the existence of `ServiceImport` based on namespace presence.

2. Service level: allow an implementation to selectively import services into specific clusters. This goes against namespace sameness and seems like it will lead to confusion. I have a service that is declared acceptable for global export in `app-ns` and I expect it to be available for consumption in `app-ns`. Some clusters have `app-ns` populated with resources, but the `ServiceImport` is not available. Now I need to check my MCS implementation's configuration to determine whether I should expect to see a ServiceImport.

(2) is slightly more flexible, but at the cost of ease of use and portability. Unless we define import selection as part of this API, it's likely that the same configuration moved to a new implementation will behave differently. (1) seems easy to reason about and would provide for consistent behavior across implementations.

I think we should start with (1) and see which use cases develop that might push us to add more flexibility